lib/puppet/provider/iis_virtual_directory/webadministration.rb

```diff
@@ -18,47 +19,71 @@ def exists?
-    cmd << "-ErrorAction Stop"
-    cmd = cmd.join
+    if is_drive_path(@resource[:physicalpath])
+      cmd = []
+      cmd << "New-WebVirtualDirectory -Name \"#{@resource[:name]}\" "
+      cmd << "-Site \"#{@resource[:sitename]}\" " if @resource[:sitename]
+      cmd << "-Application \"#{@resource[:application]}\" " if @resource[:application]
+      cmd << "-PhysicalPath \"#{@resource[:physicalpath]}\" "
+      cmd << "-ErrorAction Stop"
+      cmd = cmd.join
 
-    result   = self.class.run(cmd)
-    Puppet.err "Error creating virtual directory: #{result[:errormessage]}" unless result[:exitcode] == 0
-    @resource[:ensure] = :present
+      result   = self.class.run(cmd)
+      Puppet.err "Error creating virtual directory: #{result[:errormessage]}" unless result[:exitcode] == 0
+      @resource[:ensure] = :present
+    end
+    if is_unc_path(@resource[:physicalpath])
+      cmd = []
+      cmd << "New-Item -type VirtualDirectory \"IIS:\\Sites\\#{@resource[:sitename]}\\#{@resource[:name]}\" "
+      cmd << "-Application \"#{@resource[:application]}\" " if @resource[:application]
+      cmd << "-PhysicalPath \"#{@resource[:physicalpath]}\" "
+      cmd = cmd.join
+
+      result = self.class.run(cmd)
+      Puppet.err "Error creating virtual directory: #{result[:errormessage]}" unless result[:exitcode] == 0
+      @resource[:ensure] = :present
+    end
   end
 
   def update
+    # WORKAROUND: update() is called after destroy() by flush() in ../iis_powershell.rb
+    # New-Item -Force recreates the VirtualDirectory unless we return early.
+    return if (@resource[:ensure] == :absent)
+
     Puppet.debug "Updating #{@resource[:name]}"
 
-    if @resource[:physicalpath] and ! File.exists?(@resource[:physicalpath])
-      fail "physicalpath doesn't exist: #{@resource[:physicalpath]}"
+    verify_physicalpath
+
+    # Set-ItemProperty fails when PhysicalPath is a UNC path that is not available:
+    #   "Parameter 'PhysicalPath' should point to existing path."
+
+    if is_drive_path(@resource[:physicalpath])
+      cmd = "Set-ItemProperty 'IIS:\\Sites\\#{@resource[:sitename]}\\#{@resource[:name]}' -Name 'PhysicalPath' -Value '#{@resource[:physicalpath]}'"
+      result = self.class.run(cmd)
+      Puppet.err "Error updating virtual directory: #{result[:errormessage]}" unless (result[:exitcode] == 0 && result[:errormessage].nil?)
+    end
+    if is_unc_path(@resource[:physicalpath])
+      cmd = "Set-Item 'IIS:\\Sites\\#{@resource[:sitename]}\\#{@resource[:name]}' -type VirtualDirectory -PhysicalPath '#{@resource[:physicalpath]}' -Force"
+      result = self.class.run(cmd)
+      Puppet.err "Error updating virtual directory: #{result[:errormessage]}" unless (result[:exitcode] == 0 && result[:errormessage].nil?)
     end
 
     cmd = []
+    cmd << "Set-ItemProperty -Path 'IIS:\\Sites\\#{@resource[:sitename]}\\#{@resource[:name]}' -Name 'Site' -Value '#{@resource[:sitename]}'" if @resource[:sitename]
```

Comment:
I think this diff can be cleaned up further by not moving code that wasn't changed